django/thunderstore/community/api/experimental/views.py

```diff
@@ -0,0 +1,32 @@
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.community.api.experimental.serializers import (
+    CommunitySerializer,
+    PackageCategorySerializer,
+)
+from thunderstore.community.models import Community
+
+
+class CommunitiesExperimentalApiView(APIView):
```

Comment:
This could use `ListAPIView` instead and define `get_queryset`. Example:

```python
class PackageListApiView(ListAPIView):
    """
    Lists all communities
    """
    queryset = Community.objects.listed()
    serializer_class = CommunitySerializer
```

Reply:
When using this I get a list response instead of a JSON object with a key `communities`. Even though it being an object isn't required, it reduces breaking changes in the future. I tried making a custom serializer just for this but there doesn't seem to be a nice way to tell the `ListAPIView` that the queryset is for a specific field in the serializer.

```python
class CommunityListSerializer(serializers.Serializer):
    communities = CommunitySerializer(many=True)
```

Reply:
have you checked what's the difference with this setup and the package list? As far as I know it _should_ work as a list view when set up correclty, but I might remember something wrong

Reply:
Package list also returns just a list instead of something like `{"packages": [...]}`

Reply:
Ah, I misunderstood; I thought you wanted a flat list but didn't get one.

> Even though it being an object isn't required, it reduces breaking changes in the future.

Can you provide an example? For the experimental API this shouldn't be an issue either way, but could be relevant once it's stabilized. I don't think we want to maintain multiple different views in a single view if that's what you were implying though.

Reply:
The first example I can think of is pagination related things, but adding pagination actually converts it to an object and moves the list to a key `results`. I'm not sure what we would use it for, but there may be information we want to add relating to the list of communities. For (a not great) example, a user might want to simply know how many communities there are and not need to know anything about them. One way you could do this is return something like `{"communities": [...], "total": 42}` and even add a `limit` so they could request `0` communities and only get this metadata. This would actually be a breaking change as we currently return just `[...]` and it would be converted to an object with the results under a key.

It just seems overly restrictive to return `[...]` instead of an object as the only thing it could do is cause future changes to be breaking instead of field additions with little gain.

Reply:
Alright. As it sounds like the issue is resolved by using pagination, that sounds like the appropriate solution